File: python/lv1/두개뽑아서더하기.py
def solution(numbers):
    answerSet = set()
    for i in range(len(numbers)):
        first_num = numbers[i]
        for j in range(len(numbers)):
            second_num = numbers[j]
            if i != j:
                sum_num = first_num + second_num
                answerSet.add(sum_num)

    return list(sorted(answerSet))


numbers = [[2, 1, 3, 4, 1], [5, 0, 2, 7]]
result = [[2, 3, 4, 5, 6, 7], [2, 5, 7, 9, 12]]

for i in range(len(numbers)):
    if result[i] == solution(numbers[i]):
        print("pass")
    else:
        print("fail")

File: js/lv1/두개뽑아서더하기.js
function solution(numbers) {
  const answerSet = new Set();

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      answerSet.add(numbers[i] + numbers[j]);
    }
  }

  return Array.from(answerSet).sort((a, b) => a - b);
}

const numbers = [
  [2, 1, 3, 4, 1],
  [5, 0, 2, 7],
];

const result = [
  [2, 3, 4, 5, 6, 7],
  [2, 5, 7, 9, 12],
];

for (let i = 0; i < numbers.length; i++) {
  console.log(solution(numbers[i]));
}
